### DISCUSSION TITLE
Keeps creating 'Dedicated' world, despite different world existing, and different world named in compose file.

### DISCUSSION THREAD

**Discussion by kieron** [mbround18/valheim-docker#85] [Help!] [answered]
Hey, cant seem to figure out why every time I restart the container, it creates a world called Dedicated?

I have my compose file like this:

```version: "3"
services:
  valheim:
    image: mbround18/valheim:latest
    ports:
      - 2456:2456/udp
      - 2457:2457/udp
      - 2458:2458/udp
    environment:
      - PORT=2456
      - NAME="Palheim"
      - WORLD="Crypto"
      - PASSWORD="notmypassword"
      - TZ=Europe/London
      - AUTO_UPDATE=1
      - AUTO_UPDATE_SCHEDULE=0 3 * * *
      - PUBLIC=1
    volumes:
    - /opt/valheim/saves:/home/steam/.config/unity3d/IronGate/Valheim
    - /opt/valheim/server:/home/steam/valheim```
    - 
Thanks!

**Comment by AtroposOrbis** [comment]
Can confirm that its the auto update feature. I don't know how to correct it so just disable auto-update for now. @kieron 

@mbround18 Could you look into this for us ASAP? The auto-restart would be seriously helpful for keeping server performance up.

**Reply by mbround18** [reply]
Have y'all pulled latest recently? I did some work for auto update and validated it super early morning.

See #77 and #82

**Reply by kieron** [reply]
Hi @mbround18, when I set mine up, I simply created a `docker-compose.yml` that had what it needed, and ran that, I didn't pull this repo- is that an issue?

Or do I just need to re-run my compose file to get those new changes?

Thanks

**Reply by mbround18** [reply]
`docker-compose pull valheim`

**Reply by kieron** [reply]
Thanks I will try this shortly and see if it helps. Appreciate it.

**Comment by mbround18** [comment, marked as answer]
If you are running latest you need to pull the updates to the image occasionally.

> `docker-compose pull valheim`

**Reply by AtroposOrbis** [reply]
After pulling yesterday afternoon, the auto-update job still refuses to reopen the correct world, instead defaulting to "Valheim Server Powered by Odin".

**Reply by mbround18** [reply]
Would you mind sharing your docker compose here again?

**Reply by kieron** [reply]
I am still having issues with this btw, even after pulling. Today for example I'm seeing Incompatible version on valheim.

Thanks for all your work on this 🙂

What is the intended way to keep the game up to date? `02/26/2021 11:42:32: Peer 76561197975691701 has incompatible version, mine:0.146.8 remote 0.146.11`

**Reply by CosmicHorrorDev** [reply]
> I am still having issues with this

By "this" do you mean the world name getting messed up by auto updating?

> I'm seeing Incompatible version on valheim.

And if you server is out of date then you can

```text
$ docker exec -it --user steam $CONTAINER_NAME bash ../auto_update.sh
```

This should stop the server, create a backup, update, then start the server again.

> What is the intended way to keep the game up to date?

Currently the process for keeping the game up to date is to stop the server and try to update if `$AUTO_UPDATE` is `1` based on the cron schedule of `$AUTO_UPDATE_SCHEDULE`. The default for this is once a day at 1 am, but you can also force an update manually like above. Note that it's good to make sure you create backups just in case and there are plenty of env vars controlling behavior regarding that.
